jobs/kola-openstack.Jenkinsfile

```diff
@@ -0,0 +1,104 @@
+@Library('github.com/coreos/coreos-ci-lib') _
+
+def streams
+node {
+    checkout scm
+    streams = load("streams.groovy")
+}
+
+properties([
+    pipelineTriggers([]),
+    parameters([
+      choice(name: 'STREAM',
+             // list devel first so that it's the default choice
+             choices: (streams.development + streams.production + streams.mechanical),
+             description: 'Fedora CoreOS stream to test'),
+      string(name: 'VERSION',
+             description: 'Fedora CoreOS Build ID to test',
+             defaultValue: '',
+             trim: true),
+      string(name: 'S3_STREAM_DIR',
+             description: 'Override the Fedora CoreOS S3 stream directory',
+             defaultValue: '',
+             trim: true),
+      string(name: 'COREOS_ASSEMBLER_IMAGE',
+             description: 'Override the coreos-assembler image to use',
+             defaultValue: "coreos-assembler:master",
+             trim: true)
+    ])
+])
+
+currentBuild.description = "[${params.STREAM}] - ${params.VERSION}"
+
+def s3_stream_dir = params.S3_STREAM_DIR
+if (s3_stream_dir == "") {
+    s3_stream_dir = "fcos-builds/prod/streams/${params.STREAM}"
+}
+
+cosaPod(image: params.COREOS_ASSEMBLER_IMAGE, memory: "256Mi",
+        secrets: ["aws-fcos-builds-bot-config", "openstack-kola-tests-config"]) {
+
+    def openstack_image_filename, openstack_image_sha256, openstack_image_filepath
+    stage('Fetch Metadata/Image') {
+        shwrap("""
+        export AWS_CONFIG_FILE=\${AWS_FCOS_BUILDS_BOT_CONFIG}/config
+        cosa init --branch ${params.STREAM} https://github.com/coreos/fedora-coreos-config
+        cosa buildprep --build=${params.VERSION} s3://${s3_stream_dir}/builds
+        """)
+
+        def basearch = shwrapCapture("cosa basearch")
+        def meta = readJSON file: "builds/${params.VERSION}/${basearch}/meta.json"
+        if (meta.images.openstack) {
+            openstack_image_filename = meta.images.openstack.path
+            openstack_image_sha256 = meta.images.openstack.sha256
+            openstack_image_filepath = builds/${params.VERSION}/${basearch}/${openstack_image_filename}
+        } else {
+            throw new Exception("No OpenStack artifacts found in metadata for ${params.VERSION}")
+        }
+
+        # Copy down the openstack image from S3, verify, uncompress
```

Comment:
Minor/optional: maybe an `XXX` here for folding this into `cosa` (i.e. linking to https://github.com/coreos/coreos-assembler/issues/1508)?